### PR TITLE
fix: floating components will now get an initial position even if they are not opened

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7979

## Summary

- Removes floating-ui logic that requires a component to be open before any positioning occurs
- Updates tests
  - commonTest
  - floatingUI spec test
- Add jsdoc

### CODE REVIEW

File: packages/calcite-components/src/utils/floating-ui.ts [hunk 1]
```diff
@@ -402,29 +402,26 @@ export function getEffectivePlacement(floatingEl: HTMLElement, placement: Logica
...
- * @param options.overlayPositioning
- * @param options.placement
- * @param options.flipDisabled
- * @param options.flipPlacements
- * @param options.offsetDistance
- * @param options.offsetSkidding
- * @param options.arrowEl
- * @param options.type
- * @param delayed
+ * @param component - A floating-ui component.
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/floating-ui.ts:405]: ✨🏆✨

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 2]
```diff
@@ -1624,13 +1624,11 @@ describe("calcite-combobox", () => {
   it("should gain focus when it's items are selected via click", async () => {
     const page = await newE2EPage();
     await page.setContent(html` <calcite-combobox id="demoId">
-      <calcite-combobox-item value="test-value" text-label="test"> </calcite-combobox-item>
+      <calcite-combobox-item id="demoItemId" value="test-value" text-label="test"> </calcite-combobox-item>
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1627]: Is this test update related to the `floating-ui` changes? It seems disconnected.

Comment (jcfranco) [reply packages/calcite-components/src/components/combobox/combobox.e2e.ts:1627]: Sidebar: this test seems suspect. It's asserting on combobox receiving focus when clicking on an item when closed?

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 3]
```diff
@@ -1624,13 +1624,11 @@ describe("calcite-combobox", () => {
...
     </calcite-combobox>`);
     await skipAnimations(page);
-    const item = await page.find("calcite-combobox-item");
-    await item.click();
-    await page.waitForChanges();
     const focusedId = await page.evaluate(() => {
+      document.getElementById("demoItemId").click();
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1631]: Was the previous approach not working or why do we need to call `click` on the item?

File: packages/calcite-components/src/utils/floating-ui.spec.ts [hunk 4]
```diff
@@ -80,9 +80,9 @@ describe("repositioning", () => {
     expect(floatingEl.style.left).toBe("0");
   }
 
-  it("repositions only for open components", async () => {
+  it("repositions for unopened components", async () => {
     await reposition(fakeFloatingUiComponent, positionOptions);
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/floating-ui.spec.ts:84]: For clarity, could you call `assertPreOpenPositioning` at the beginning of each of these tests?

**Review by jcfranco — APPROVED** [review]
Just had a question on combobox E2E test updates, but this LGTM!

Thanks for updating the util doc! 🏆

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/combobox/combobox.e2e.ts:1627]: @jcfranco yeah I couldn't get this test to pass otherwise. Not sure if its related or not but your observations make sense. I can try setting it to be open?

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts (hunk 3 above)
Comment (driskull) [reply packages/calcite-components/src/components/combobox/combobox.e2e.ts:1631]: Test wasn't passing. This is the only test that was failing
